File: src/main/java/shopping/dto/ProductResponse.java
package shopping.dto;

import shopping.domain.product.Product;

import java.math.BigDecimal;

public record ProductResponse(
        Long id,
        String name,
        BigDecimal price,
        String imageUrl
) {

    public static ProductResponse from(Product product) {
        return new ProductResponse(
                product.getId(),
                product.getName(),
                product.getPrice(),
                product.getImageUrl()
        );
    }
}

File: src/main/kotlin/shopping/product/presentation/dto/response/ProductResponse.kt
package shopping.product.presentation.dto.response

import shopping.product.domain.Product
import java.math.BigDecimal

class ProductResponse(product: Product) {
    val id: Long = product.id
    val sellingPrice: BigDecimal = product.sellingPrice
    val fixedPrice: BigDecimal = product.fixedPrice
    val productName: String = product.productName
    val productAmount: Int = product.productAmount
    val productImage: String = product.productImage
    val productDescription: String = product.productDescription
}
